### ISSUE TITLE
Warning. Dropbox Public Folders Will Soon Be Private.

### ISSUE THREAD

**Issue by bfayers** [skycocker/chromebrew#268]
https://www.dropbox.com/help/16

However if you upload to any other directory, right click hit share then hit create link then copy link and change the end to ?dl=1 it still works.

Bit of a non-issue just a warning because I know lots are in the public folder.

**Comment by thedamian** [comment]
This might bring back the talk we've had before of maybe (at least with the initial packages) maybe should be hosted HERE on github. 

What does everyone else say?

**Comment by bfayers** [comment]
I think we should. Or perhaps something like [https://bintray.com/](Bintray). Would be interesting to have that. or use Maven and have another repo (chromebrew-sources) that compiles automatically from source with auto updates. Again a suggestion.

**Comment by minektur** [comment]
github, bintray, or a cheap shared hosting site (can you put binary pacakges on google-sites? - I could set up a g-suite/google-sites page for any of several domains I administer and then give access to whoever needs...)  

From this list though github seems the obvious choice - we wouldn't be adding another dependency.

**Comment by thedamian** [comment]
This is a conversation we've done before.

I've setup a bintray and I think I made it public so others can share: https://bintray.com/thedamian/ChromeBrew

BUT I'm thinking github would make MUCH more sense for me.

**Comment by bfayers** [comment]
github sounds like the best choice really. GIT-LFS is interesting

**Comment by jam7** [comment]
Same here,.  I prefer github, but I couldn't figure out how to put multiple binaries like gcc, glibc, etc. in release page.

**Comment by minektur** [comment]
I just made a test repo with a test release of a bunch of binaries.  I had to manually predict what the tag would be so I could embed a url for one of the binaries in some source code ahead of time (e.g. pre-tag).  There is probably some way to automate it.

See

https://github.com/minektur/test-binary-release


When you make the release, there is a place you can drag-n-drop binaries to be included on the release page.  when you "publish" the release it tags the source code at that moment and creates/publishes the release.

Try it a couple of times in a test repo - not too bad.  and then we can host all the content here on github.  Of course I'm not in charge :)  github would work and we already have this page here...

**Comment by jam7** [comment]
I've just checked https://github.com/minektur/test-binary-release/releases.  It's nice.

**Comment by bfayers** [comment]
@minektur If I, or someone else was to make a repo called something like chromebrew-packages and give a good few people merge possibility this could work. Just need a way to automate the "Releases" thing.

**Comment by cstrouse** [comment]
We may also want to look at using a tool like gitian so users can trust that the binaries are in fact safe to use. Safety of the binaries from homebrew has been brought up in a bunch of discussions about Homebrew.

**Comment by minektur** [comment]
I'd be happy to set one up... but it would be better if the repo were connected with/part of this one.  

paging @skycocker  - you are summoned - can you make us a repo and give out some permissions?

**Comment by thedamian** [comment]
I can probably do it. You mean a new branch roght? Not a whole new repo?

**Comment by bfayers** [comment]
I think repo, because I dont think you can make releases of a branch

**Comment by finwo** [comment]
If having a hosting location is the issue, there is already an issue proposing mirrors: #162

**Comment by jam7** [comment]
It looks like everyone can make release repository once we forked this chromebrew.  The questions is how to manage them.

For example, one or few collaborators make their own repository and put binaries there each time they merge new packages with binary files.  For example, each time someone makes a PR with dropbox URL, it is required to download binaries and make release page in each collaborator's repository and modify URL in the PR.

Isn't it too much works?  I'm pushing many binaries often recently, so... :)

If it is possible to use GIT-LFS, it may reduce the amount of works and automate it somehow.  Probably it requires to modify crew or wget to handle GIT-LFS, tho.

**Comment by jam7** [comment]
Ok.  GitHub release page wasn't as hard as what I expected.

I made two scripts to make maintenance easy.  I put them in #343.  By using these scripts, it is possible to retrieves newly provided binary file URLs and upload them to certain release page by:
```
<<create release page first and give it a TAG>>
$ sh create_url_list.sh > url_list
$ wget -P bin -i url_list
$ sh upload_github_release.sh TAG bin/*
```
I made test page at https://github.com/jam7/chromebrew/releases I made like above.  I can do it on my repository only.  I wish collaborator performs the similar things here...
